pydantic/dataclasses.py

```diff
@@ -408,6 +408,8 @@ def create_pydantic_model_from_dataclass(
     model.__doc__ = dc_cls_doc if dc_cls_doc is not None else dc_cls.__doc__ or ''
     return model
 
+def _is_field_cached_property(obj: 'Dataclass', k: str) -> bool:
+    return isinstance(getattr(type(obj), k, None), cached_property)
```

Comment:
This works. We could also use `typing.get_type_hints` for this. Not sure which is preferred. Thoughts @hramezani?

Reply:
The function itself is easy to understand. So, I don't have strong opinion to change.
Also, this is V1 and it will not going to be maintained more.